crates/sema/src/lib.rs

```diff
@@ -100,6 +101,32 @@ pub fn parse_and_resolve(pcx: ParsingContext<'_>) -> Result<()> {
     Ok(())
 }
 
+/// Parses and lowers to HIR, recursing into imports.
+pub fn parse_and_lower_to_hir<'hir>(
```

Comment:
I want to avoid any code dup here and also avoid using Hir directly, but instead through gcx; I think what we can do here is something like this
```rust
fn parse_and_resolve() {
    ...
    let global_context = parse_and_lower()?;
    analysis(gcx)?;
    Ok(())
}

fn parse_and_lower<'hir>(pcx: '_, hir_arena: 'hir) -> impl Deref<Target=Gcx<'hir>> {
    // all the code currently in parse_and_resolve ...

    GcxWrapper(...)
}

struct GcxWrapper(GlobalContext);
impl Deref for GcxWrapper {
    // move the unsafe here
}
impl Drop for GcxWrapper {
    // move from OnDrop closure
}
```

This way we can keep the drop impl and avoid a closure for using Gcx

Reply:
@DaniPopes would the approach in https://github.com/paradigmxyz/solar/pull/210/commits/9c654ccb33690e5664274e324dadf80e779aa369 which doesn't introduce a new wrapper work?